frontend/src/metabase/visualizations/visualizations/PivotTable.jsx

```diff
@@ -196,6 +194,151 @@ export default class PivotTable extends Component {
       );
     }
 
+    // Create three memoized cell renderers
+    // These are tied to the `multiLevelPivot` call, so they're awkwardly shoved in render for now
+
+    const topHeaderRenderer = _.memoize(
+      ({ key, style, columnIndex }) => {
+        if (columnIndex === topIndex.length) {
+          return (
+            <div
+              key={key}
+              style={style}
+              className="flex-column justify-end px1 pt1"
+            >
+              <Cell value={t`Row totals`} width={valueIndexes.length} />
+            </div>
+          );
+        }
+        const rows = topIndex[columnIndex];
+        return (
+          <div key={key} style={style} className="flex-column px1 pt1">
+            {rows.map((row, index) => (
+              <Flex style={{ height: CELL_HEIGHT }}>
+                {row.map(({ value, span }) => (
+                  <div
+                    style={{ width: CELL_WIDTH * span }}
+                    className={cx({
+                      "border-bottom": index < rows.length - 1,
+                    })}
+                  >
+                    <Ellipsified>
+                      {index < topIndexFormatters.length
+                        ? topIndexFormatters[index](value)
+                        : value // Metric names don't have formatters
+                      }
+                    </Ellipsified>
+                  </div>
+                ))}
+              </Flex>
+            ))}
+          </div>
+        );
+      },
+      ({ columnIndex }) => columnIndex,
+    );
+
+    const leftHeaderRenderer = _.memoize(
+      ({ key, style, index }) => {
+        if (index === leftIndex.length) {
+          return (
+            <Flex key={key} style={style}>
+              <Flex flexDirection="column">
```

Comment:
Do we need the nested flex element here or could we just set `flexDirection="column"` on the outer wrapper and eliminate another node?

Reply:
Good call! Both `<Flex>` elements were unnecessary. I just set `key` and `style` on `<Cell>`.